var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -56,12 +63,65 @@ class Faiss(AutotoolsPackage, CudaPackage):
     # also, some include paths in gpu/tests/Makefile are missing
     patch("fixes-in-v1.6.3.patch", when="@1.6.3")
 
+    # patch for v1.7.2
+    # a shared object is missing in the python/setup.py
+    #   https://github.com/facebookresearch/faiss/issues/2063
+    #   https://github.com/facebookresearch/faiss/pull/2062
+    # a header is missing in a test file
+    #   https://github.com/facebookresearch/faiss/issues/2300
+    patch("fixes-in-v1.7.2.patch", when="@1.7.2")
+
+    def setup_run_environment(self, env):
+        if "+python" in self.spec:
+            env.prepend_path("PYTHONPATH", python_platlib)
+
+
+class PythonPipBuilder(spack.build_systems.python.PythonPipBuilder):
```

Comment:
This one is used as part of another object, which is fine. Maybe it's better scoped as a local class defined in the `CMakeBuilder.install` method?